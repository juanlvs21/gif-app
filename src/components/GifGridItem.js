import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ url, title }) => {
  return (
    <div className="card animate__animated animate__bounceIn">
      <img src={url} alt={title} />
    </div>
  );
};

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
