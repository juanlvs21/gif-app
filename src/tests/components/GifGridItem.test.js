import { shallow } from "enzyme";

import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en el componente GifGridItem", () => {
  const title = "Title example";
  const url = "http://example/image.png";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Cargar componente GifGridItem correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("Debe tener una imagen con un src y un alt igual a los props", () => {
    const img = wrapper.find("img");

    // expect(img.props().src).toBe(url);
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener animate__bounceIn", () => {
    const div = wrapper.find("div");

    expect(div.prop("className").includes("animate__bounceIn")).toBeTruthy();
  });
});
