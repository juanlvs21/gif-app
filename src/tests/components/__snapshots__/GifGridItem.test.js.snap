// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Cargar componente GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Title example"
    src="http://example/image.png"
  />
  <p>
    Title example
  </p>
</div>
`;
